engineapi/backup_monitor.go

```diff
@@ -70,19 +70,17 @@ func NewBackupMonitor(logger logrus.FieldLogger,
 	}
 
 	// Call engine API snapshot backup
-	if backup.Status.State == longhorn.BackupStateNew {
```

Comment:
I don't understand why we need to remove this check.
- When the backup creation is not started yet, this check won't block the creation.
- When the backup creation is already in progress, the below invocation `engineClient.SnapshotBackup` would return [a (locking) error](https://github.com/longhorn/longhorn-engine/blob/cda02873ef0789a56fcca9ffcdbc5bea76f09018/vendor/github.com/longhorn/backupstore/deltablock.go#L125) during [the backup init](https://github.com/longhorn/longhorn-engine/blob/cda02873ef0789a56fcca9ffcdbc5bea76f09018/pkg/backup/main.go#L127). Since the 1st step of the backup creation is [picking up a random replica](https://github.com/longhorn/longhorn-engine/blob/cda02873ef0789a56fcca9ffcdbc5bea76f09018/pkg/sync/backup.go#L60-L65) rather than checking all replicas and see if there is a replica processing the creation.